jupyterhub_singleuser_profiles/api/api.py

```diff
@@ -49,7 +49,7 @@ def decorated(*args, **kwargs):
                 user = auth.user_for_token(token, use_cache=False)
         else:
             user = None
-        if for_user and user.get('admin'):
+        if for_user and user and user.get('admin'):
             user['name'] = for_user
             user['admin'] = False
         if user:
```

Comment:
I think it would make sense to keep the condition as is and just move it under the `if user` condition on line 55

Reply:
Good catch.  I moved the `for_user` check and assignment under the `if user` check